### DISCUSSION TITLE
Timezone in Area Chart

### DISCUSSION THREAD

**Discussion by smurraystrata** [Budibase/budibase#11304] [Help] [answered]
Hello all! 

I have an automation that pulls data every 5 minutes and adds the date and time to a Date and Time field in a table (along with other things). This works excellently and the tables in the design section look great. I recently decided to toss in (love how easy that is to do with Budibase) a quick Area Graph which helps visualize the data, but when using the Date and Time as the Label column it shows UTC instead of our timezone, while the table below the graph uses the current timezone of the browser accessing it. Are there any suggestions to fix that? The table this is being pulled from is not set to ignore time zones.

The two times circled in purple on my screenshot should both show 13 for the hour.
![image](https://github.com/Budibase/budibase/assets/120056389/43c616e8-fe5c-4537-aa92-971ec5b24568)

Thank you for the help!

**Comment by melohagan** [comment, marked as answer]
Hey @jedman20 

You could try adding a [Formula](https://docs.budibase.com/docs/formula) column to your table that uses JavaScript to apply an offset to make the time display as it does on the front-end.

Then you could use that formula column as your chart label column.

**Reply by Treysen-Zobell** [reply]
I tried this, it doesn't work like you'd expect. It changes the order left newest, right oldest, and gives an individual tick for each day/time entry instead of a linear scale. I'd guess that the problem originates in that the calendar column is reliably interpretable as a number so it can be scaled, while the formula is treated as text.

**Reply by melohagan** [reply]
You can return a number from a formula column by using JavaScript bindings. Handlebars will always be text, but JavaScript bindings can handle different data types

**Comment by jcyh0120** [comment]
Even if javascript return a Date object, the tick still show whole the string text. Still can't find a way to make the label from formula column act like a date column.

<img width="715" alt="截圖 2024-04-03 下午5 04 22" src="https://github.com/Budibase/budibase/assets/4725814/cd375777-180d-4946-8a66-ac71401ddd85">


> Hey @jedman20
> 
> You could try adding a [Formula](https://docs.budibase.com/docs/formula) column to your table that uses JavaScript to apply an offset to make the time display as it does on the front-end.
> 
> Then you could use that formula column as your chart label column.

**Reply by smurraystrata** [reply]
Hello @jcyh0120,

That is actually what we ended up doing.

```
let date = new Date($("Date and Time"))
let tzCorrectedDate = new Date((typeof date === "string" ? new Date(date) : date).toLocaleString("en-US", {timeZone: "mst"}));
tzCorrectedDate.setHours(tzCorrectedDate.getHours() + 0)

let month = tzCorrectedDate.toLocaleString('default', { month: 'short' })
let day = tzCorrectedDate.getDate()
let year = tzCorrectedDate.getFullYear()
let hours = ("0" + tzCorrectedDate.getHours()).slice(-2)
let minutes = ("0" + tzCorrectedDate.getMinutes()).slice(-2)

return month + " " + day + " " + year + ", " + hours + ":" + minutes
```

**Reply by jcyh0120** [reply]
This works as a fixed label tick. However, I saw the original label using Date column will change the tick when I use "1day", "7days", etc to filter the data, which is realllyy great for viewing the chart. I still cannot find a direct way to achieve the same behavior.
